### DISCUSSION TITLE
Check free space before starting

### DISCUSSION THREAD

**Discussion by John-Gee** [DeFiCh/app#854] [Ideas]
Since a bunch of people have had issues with not having enough free space, but not realizing it by themselves and instead wasting time (and maybe data from their cap) on trying to sync, I think it'd be nice to have a check for that, before downloading (whether it's the first download, or just to get to current).

Not sure if the app now knows how much space it'll need, but you probably could get an estimate from how big a block is and how many there are, or from the size of the snapshot and how far it is from current. Then of course add some extra for buffering.

Maybe this could work like an installer, and offer to use another partition if there is free space there?

@thedoublejay Sorry, I said I wouldn't bother you again, but maybe this isn't too hard and yet it may be quite useful to new people.

Aside from the wallet, I think it'd be wise to have that information on the sites too, ie on https://defichain.com/downloads/ at least, and maybe on here too. Not sure who to ping for that though.

**Comment by OKnr66** [comment]
It is absolutely important to query the disk space during installation for such installation sizes. You can also build in a termination condition that is not installed, if after the fullnode installation less than 30 GB on the disk are free, then abort or suggestion for outsourcing, if possible.
Ich stimme zu.
